components/datetime/src/mock/time_zone.rs

```diff
@@ -62,6 +64,48 @@ impl MockTimeZone {
             time_variant,
         }
     }
+
+    /// Overwrite the metazone id in MockTimeZone.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::datetime::date::GmtOffset;
+    /// use icu::datetime::metazone::MetaZoneCalculator;
+    /// use icu::datetime::mock::time_zone::MockTimeZone;
+    /// use icu_calendar::DateTime;
+    /// use icu_datetime::provider::time_zones::{MetaZoneId, TimeZoneBcp47Id};
+    /// use icu_locid::locale;
+    /// use tinystr::tinystr;
+    ///
+    /// let provider = icu_testdata::get_provider();
+    /// let mzc = MetaZoneCalculator::new(locale!("en"), &provider);
+    /// let mut tz = MockTimeZone::new(
+    /// GmtOffset::default(),
+    ///     /* time_zone_id */ Some(TimeZoneBcp47Id(tinystr!(8, "gugum"))),
```

Comment:
Issue: We need to discuss how one goes about constructing a `MockTimeZone`. Here, you are constructing an invalid one, because "gugum" should likely have an offset of GMT+10 or similar. CC @nordzilla @Manishearth

Reply:
This question is easier for full datetimes because much of the information is populated automatically from the data providers.

As far as stand-alone TimeZone objects, it's a lot trickier. I'm not sure that we should be verifying whether a stand-alone TimeZone object is valid. 

Or perhaps that could be an opt-in sort of check? To do such validations would require loading historical data, cross checking to make sure that this GMT offset may have made sense at some point in history with this zone. 

I would lean toward the stance that when people are creating stand-alone time-zone objects, it's not our purpose to attempt to validate it. 

When these objects are created from zoned datetimes, everything should be valid according to what is provided by the data providers.

Reply:
Decision from discussion with @sffc @nordzilla @Manishearth @zbraniecki:

- Make offset optional
- Add a field to the data struct with the last-last-resort fallback, which will generally be "GMT+?"
- When using this last-last-resort fallback, throw a debug assertion

Reply:
I am not clear that how to use the offset fallback in `MockTimeZone`, we can discuss in the meeting.